Privacy settings

etoledom opened the pull request:
Closes GRA-738

This PR implements the Privacy Settings screen.
The Analytics switch will deactivate sending analytics to the backend (and will be logged as `🔹 Tracking (Locally):`.
The Crash Report switch will be linked after https://github.com/Automattic/Gravatar-iOS/pull/103 is merged.

A new analytic event has been added:
- `about_privacy_settings_button_tapped`

And 
- `about_privacy_policy_button_tapped`
Has been replaced by:
- `privacy_settings_privacy_policy_button_tapped`

<img width="40%" alt="CleanShot 2025-08-15 at 18 33 24@2x" src="https://github.com/user-attachments/assets/17053a78-3e3e-4f58-a821-bcd404ed6832" />

### To test:
- Open the Main Menu
- Tap on `About this app`
- Tap on `Privacy Settings`
  - Check that the Privacy Settings screen looks correct
  - Check that the switches are ON by default
- Switch off  `Share Analytics Data`
- Trigger some events
  - Check that they are logged as `Tracking (Locally)`
- Restart the app
  - Check that the events are still logged as `Tracking (Locally)`
- Go back to `Privacy Settings`
  - Check that the  `Share Analytics Data` switch is Off
- Turn the switch back ON.
  - Check that the events are tracked as normal.

etoledom committed:
Adding privacy settings screen

etoledom committed:
Format

etoledom committed:
Add snapshot tests

etoledom committed:
Remove unused analytics event

etoledom committed:
Update strings in base locale

etoledom committed:
Update snapshots

etoledom committed:
Reorganize code

etoledom committed:
Improve dark theme colors

etoledom committed:
Merge branch 'trunk' into etoledom/privacy-settings

pinarol commented on GravatarApp/CommonViews/MainMenu/PrivacySettings/PrivacySettingsScreen.swift:
```diff
@@ -0,0 +1,135 @@
+    )
+    static let crashReportMessage = NSLocalizedString(
+        "PrivacySettings.CrashReport.description",
+        value: "Share information with our analytics tool about how you interact with this app and our services.",
```
Looks like a copy paste error to use the same text for crash reporting as well?

pinarol commented:
The crash reporting PR is merged, you might want to update from trunk.

pinarol commented:
Not necessarily in this PR but we might want to refactor the About screen. We can have a separate Settings screen and move Privacy settings and account deletion to there 🤔. At the moment About screen outgrew its purpose...

etoledom committed:
Update crash report message text

etoledom commented:
> Not necessarily in this PR but we might want to refactor the About screen. We can have a separate Settings screen and move Privacy settings and account deletion to there 🤔. At the moment About screen outgrew its purpose...

💯 Agree.
But this is a structural app design change. Not just an About screen refactor.

Updated from trunk and updated the wrong text 👍 

I've been working on the integration of crash reporting with this screen and is a big diff, so I prefer to merge this one and have that integration in a different PR.

etoledom committed:
Format

etoledom committed:
Update snapshots

etoledom committed:
Merge branch 'trunk' into etoledom/privacy-settings

pinarol reviewed (approved):
Cool, let's go! 🎉
